apps/studio/src/components/TabQueryEditor.vue

```diff
@@ -31,13 +30,16 @@
           >Close Tab</a>
         </div>
       </div>
-      <textarea
-        name="editor"
-        class="editor"
-        ref="editor"
-        id=""
-        cols="30"
-        rows="10"
+      <sql-text-editor
+        v-model="unsavedText"
+        :lang="connectionType"
+        :extra-keybindings="keybindings"
+        :vim-config="vimConfig"
+        :height="editorHeight"
+        :read-only="editorReadOnly"
+        @initialized="handleEditorInitialized"
+        @cursorActivity="handleCursorActivity"
+        @interface="editorInterface = $event"
```

Comment:
No, it's not CodeMirror specific. It's an interface for the `TextEditor` component. The idea is to avoid calling child functions directly, but the child provides an interface to the parent so later the parent should call anything that's available on that interface instead. I got this idea from this [answer](https://stackoverflow.com/a/70723343/10012118).

I'm trying to avoid `props/watcher` method, cause this can get so messy IMO. I think we can do event bus? `this.$emit()` and then `this.registerHandlers`? We don't have to put the listener names to the `AppEvent` though.

Reply:
I think I agree `editorInterface` is not the best. Maybe just do event bus.